woocommerce/woocommerce-ios
# Reviews UI mark III. Mark reviews as read

Fixes #1252. 

Third step in the implementation the Reviews tab: mark reviews as read.

The logic is as described in p99K0U-1Fk-p2. In a nutshell, we would mark as read the notification matching the review (where the review id is the notification's comment id)

Reviews should we be marked as read by navigating to the review details screen. Also, all reviews can be marked as read by tapping the top right button (check mark) in the reviews screen.

<img src="https://user-images.githubusercontent.com/2722505/65593675-956bd680-dfc3-11e9-84f4-e69611cc6309.gif" width="350"/>

## Changes
- Add logic to mark a single review as read when entering the ReviewDetailsViewController
- Add logic to mark all reviews as read in ReviewsViewModel
- Add logic to fetch notifications and observe them in ReviewsViewModel

## Testing
As with other PRs in this series...
- Checkout the branch, run the tests
- Undo the feature flag. Because the NotificationsViewController is instantiated in a Storyboard, it will be necessary to touch `FeatureFlag.swift` to:
```
        case .reviews:
            return BuildConfiguration.current == .localDeveloper
```
and it will also be necessary to change the class of the first view controller in the Notifications.storyboard from NotificationsViewController to ReviewsViewController
- As an alternative, revert locally the commit dc49cdf1e3b57d3b926923268393d73c32a64319
- Navigate to the reviews tab > select a review. The review should be marked as read
- Navigate to the reviews tab. Tap the "mark all as read" button. All reviews should be marked as read


Update release notes:
